python/neuroglancer/pipeline/storage.py

```diff
@@ -100,27 +65,45 @@ def extract_path(cls, layer_path):
         else:
             return cls.ExtractedPath(*match.groups())
 
-    def put_file(self, file_path, content, compress=True):
+    def put_file(self, file_path, content, compress=False):
         """ 
         Args:
             filename (string): it can contains folders
             content (string): binary data to save
         """
-        if compress:
-            content = self._compress(content)
+        return self.put_files([ (file_path, content) ], compress, block=False)
 
-        if self._n_threads:
-            # None is the non-existant callback
-            self._queue.put(('put_file', None, file_path, content, compress), block=True)
-        else:
-            self._interface(self._path).put_file(file_path, content, compress)
-        return self
+    def put_files(self, files, compress=False, block=True):
+        """
+        Put lots of files at once and get a nice progress bar. It'll also wait
+        for the upload to complete, just like get_files.
+
+        Required:
+            files: [ (filepath, content), .... ]
+        """
+        def base_uploadfn(path, content, interface):
+            interface.put_file(path, content, compress)
```

Comment:
The inconsistency between `put_files` and `get_files` is due to the fact that we can't control the other end of the connection, but we can control this side to achieve more flexibility.